products/bridge/smart-contracts/contracts/periphery/BridgedToken.sol

```diff
@@ -29,7 +49,32 @@ contract BridgedToken is IERC20, ERC20, ERC20Burnable, Ownable {
         _mint(to, amount);
     }
 
+    function setLockProxyAddress(address lockProxyAddress_) external onlyOwner {
+        lockProxyAddress = lockProxyAddress_;
+    }
+
+    function transfer(
+        address to,
+        uint256 value
+    ) public override(ERC20, IERC20) returns (bool) {
+        mintIfLockProxy(msg.sender, to, value);
+        return super.transfer(to, value);
+    }
+
+    function transferFrom(
+        address from,
+        address to,
+        uint256 value
+    ) public override(ERC20, IERC20) returns (bool) {
+        mintIfLockProxy(msg.sender, to, value);
```

Comment:
Why is  mintIfLockProxy() here?  AFAICT all this does is to mint sufficient tokens to satisfy the transfer if the sender is the lock proxy, regardless of the from address. I can't see why you'd do that?

Reply:
Also the reason I could not just override `_transfer` is because ERC20 on Openzeppelin's v5 does not mark `_transfer` as virtual anymore. So I have to manually override the `transfer` and `tranferFrom` functions

Reply:
Ah! Good point well made - understood. Thanks!